tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/stats.py

```diff
@@ -0,0 +1,127 @@
+from typing import Union
+
+import numba
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import scipy.sparse as sparse
+
+from .globals import CENSUS_OBS_STATS_COLUMNS, CENSUS_VAR_STATS_COLUMNS
+
+
+def get_obs_stats(
+    raw_X: Union[sparse.csr_matrix, sparse.csc_matrix],
+) -> pd.DataFrame:
+    """Compute summary stats for obs axis, and return as a dataframe."""
+
+    if not isinstance(raw_X, sparse.csr_matrix) and not isinstance(raw_X, sparse.csc_matrix):
+        raise NotImplementedError(f"get_obs_stats: unsupported type {type(raw_X)}")
```

Comment:
isn't allowing csr a performance concern? and vice versa for get_var_stats().

Reply:
I don't understand the question.  What sort of performance concern?

The data is natively in either CSR or CSC as that is what AnnData mandates.  Converting to COO or some other format would not be free.

Reply:
👍 I see, although the CSR is the preferred type here, the caller has whatever it has...